src/java/org/apache/cassandra/audit/AuditLogManager.java

```diff
@@ -135,7 +143,21 @@ else if (e instanceof AuthenticationException)
             builder.setType(AuditLogEntryType.REQUEST_FAILURE);
         }
 
-        builder.appendToOperation(QueryEvents.instance.getObfuscator().obfuscate(e.getMessage()));
+        // A syntax error may reveal the password in the form of 'line 1:33 mismatched input 'secret_password''
+        String safeMsg = e.getMessage();
+        if (e != null && e instanceof SyntaxException && queries != null && !queries.isEmpty())
+        {
+            for (String query : queries)
+            {
+                if (query.toLowerCase().contains(PasswordObfuscator.PASSWORD_TOKEN))
+                {
+                    safeMsg = "Syntax Exception. Obscured for security reasons.";
+                    break;
+                }
+            }
+        }
+
+        builder.appendToOperation(PasswordObfuscator.obfuscate(safeMsg));
```

Comment:
I would have use extracted the logic creating the message into:
```
public String obfuscatePasswordInformation(Exception e, List<String> queries)
{
    if (e instanceof SyntaxException && queries != null && !queries.isEmpty())
    {
        for (String query : queries)
        {
            if (query.toLowerCase().contains(PasswordObfuscator.PASSWORD_TOKEN))
            {
                return "Syntax Exception. Obscured for security reasons.";
            }
        }
    }

    return PasswordObfuscator.obfuscate(e.getMessage());
}
```

Reply:
I'll take that one as optional if you don't mind as I prefer it inline. Total personal thing of mine lol